pyro/distributions/empirical.py

```diff
@@ -151,7 +163,7 @@ def mean(self):
                              "or ``torch.float64``. If these are samples from a " +
                              "`Categorical` distribution, consider converting to a " +
                              "`OneHotCategorical` distribution.")
-        return self._weighted_mean(self._samples)
+        return self._weighted_mean(self._flatten_chain(self._samples))
```

Comment:
Note that to keep the implementation simple, this simply gives the `mean`/`variance` across all chains. If the user needs to implement chain specific mean they can get all samples via `.enumerate_support(flatten=False)`